neuraxle/checkpoints/base_checkpoint_step.py

```diff
@@ -0,0 +1,57 @@
+from neuraxle.base import BaseStep
+
+
+class BaseCheckpointStep(BaseStep):
```

Comment:
Add ABC for making it an abstract class: 
`class BaseCheckpointStep(BaseStep, ABC):`